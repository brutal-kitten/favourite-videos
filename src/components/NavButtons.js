import React, { Component } from 'react';
import ShowDemo from './ShowDemo';
import DeleteList from './DeleteList';
import ShowFavorite from './ShowFavorite'
import Sort from './Sort'

class NavButtons extends Component {
  render() {

    return (
      <div className="navButtons grid-container2">
        <ShowDemo
          showDemo={this.props.showDemo}
        />
        <DeleteList
          deleteList={this.props.deleteList}
        />
        <ShowFavorite
          showFavorite={this.props.showFavorite}
        />
        <Sort
          sort={this.props.sort}
        />
      </div>
    )
  }
}
export default NavButtons;
