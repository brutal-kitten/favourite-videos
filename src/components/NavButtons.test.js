import React from 'react';
import ReactDOM from 'react-dom';
import NavButtons from './NavButtons.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <NavButtons />).toJSON();
  expect(tree).toMatchSnapshot();
});

const mockShowDemo = jest.fn();
const mockDeleteList = jest.fn();
const mockReturnToList = jest.fn();
const mockShowFavorite = jest.fn();
const mockSort = jest.fn();


describe('when showFav is true', () => {
  const props = {
    showFav: true,
    showDemo: mockShowDemo,
    deleteList: mockDeleteList,
    returnToList: mockReturnToList,
    showFavorite: mockShowFavorite,
    sort: mockSort
  }
  const wrapper = mount(<NavButtons {...props}/>);

  it('should have "return to list" button', () => {
    expect(wrapper.find('.returnToList').exists()).toBeTruthy();
  });

  it('should not  have "show favorite" button', () => {
    expect(wrapper.find('.showFav').exists()).toBeFalsy();
  });

});

describe('when showFav is false', () => {

  const props2 = {
    showFav: false,
    showDemo: mockShowDemo,
    deleteList: mockDeleteList,
    returnToList: mockReturnToList,
    showFavorite: mockShowFavorite,
    sort: mockSort
  }

  const  wrapper2 = mount(<NavButtons {...props2}/>);

  it('should not have "return to list" button', () => {
    expect(wrapper2.find('.returnToList').exists()).toBeFalsy();
  });

  it('should have "show favorite" button', () => {
    expect(wrapper2.find('.showFav').exists()).toBeTruthy();
  });
});
