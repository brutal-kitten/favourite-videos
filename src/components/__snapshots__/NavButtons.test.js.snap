// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="navButtons grid-container2"
>
  <div
    className="showDemo"
  >
    <button
      className="buttonShowDemo"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Show demolist
    </button>
  </div>
  <div
    className="deleteList"
  >
    <button
      className="buttonDeleteList"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Delete list
    </button>
  </div>
  <div
    className="showFav"
  >
    <button
      className="buttonShowFav"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Show favorite
    </button>
  </div>
  <div
    className="sort"
  >
    <span
      className="sortText"
    >
      Sort by
    </span>
    <select
      id="select"
      onChange={[Function]}
      tabIndex="0"
      value="new"
    >
      <option
        value="new"
      >
        Newest added
      </option>
      <option
        value="old"
      >
        Oldest added
      </option>
    </select>
  </div>
</div>
`;
